### PR TITLE
Add V2 Vulnerability Filtering API, fix call to get session metadata

### PR DESCRIPTION
**Pull request by atzawada**
This change implements part of the V2 Application Vulnerability Filtering API, specifically this endpoint:
```
POST  /ng/{orgUuid}/traces/{appId}/filter
```

The other change is to the way the URL is generated for getting the session metadata for an application. Previously the URL would not be generated correctly if the `TraceFilterForm` had no filters or was null.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (bryanateoan) [inline .vscode/settings.json:1]: Please remove this file.

We started adding any IDE specific files into `.gitignore`

File: src/main/java/com/contrastsecurity/models/TraceFilterBody.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package com.contrastsecurity.models;
```
Comment (bryanateoan) [inline src/main/java/com/contrastsecurity/models/TraceFilterBody.java:1]: Any reason we can't reuse `TraceFilterForm` and just add the missing fields?

File: src/main/java/com/contrastsecurity/sdk/ContrastSDK.java [hunk 3]
```diff
@@ -761,6 +761,35 @@ public Traces getTraces(String organizationId, String appId, TraceFilterForm for
     }
   }
 
+  /**
```
Comment (bryanateoan) [inline src/main/java/com/contrastsecurity/sdk/ContrastSDK.java:764]: If we reuse the `TraceFilterForm` we can also just update the existing `getTraces` method to use V2 vulnerability filtering. Both methods are getting a list of vulnerabilities based on a filter.

**Review by bryanateoan — CHANGES_REQUESTED** [review]
Hi Andy, thank you very much for the contribution.

I have a few request and questions.
1. Remove vscode settings file
2. Questions about re-using TraceFilterForm

**Comment by atzawada** [comment]
Hi Brian, is there any reason to maintain the fields on the old v2 TraceFilterForm? Are they used to query in a similar fashion anywhere else in the v3 API?

**Comment by bryanateoan** [comment]
TraceFilterForm is used by `getSessionMetadataForApplicationUrl` and `getTracesByOrganization`. However, similar to getTraces, it is also moving to having its parameters in the body. The `TraceMetadataFilter` can probably be encoded similar to how `appVersionTags` is encoded for the toQuery method.

**Comment by bryanateoan** [comment]
@atzawada, I'm okay with adding a new getTraces method. Can you mark the existing getTraces that uses the TraceFilterForm as `@Deprecated`.

**Review by bryanateoan — APPROVED** [review]
Thank you very much for the contribution!

**Pull request merged**